pallets/gear/src/lib.rs

```diff
@@ -1202,6 +1226,13 @@ pub mod pallet {
 
             QueueOf::<T>::queue(dispatch).map_err(|_| Error::<T>::MessagesStorageCorrupted)?;
 
+            if expiration_block > block_number {
```

Comment:
Condition is always true
```
CURRENT_BLOCK + 1000 > CURRENT_BLOCK
```

Reply:
Nope, free period may be changed to zero after runtime upgrade.